main/src/init/fixed_boundaries.hpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2021 CSCS, ETH Zurich
+ *               2021 University of Basel
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*! @file flag particles for fixed boundary conditions
+ *
+ * @author Lukas Schmidt
+ */
+
+#pragma once
+
+#include "cstone/sfc/box.hpp"
+#include "sph/particles_data.hpp"
+
+namespace sphexa
+{
+template<class T, class Dataset>
+void applyFixedBoundaries(Dataset& d, cstone::Box<T>& box)
```

Comment:
* This should not take the whole Dataset, just one out of (x,y,z) and all of (vx, vy, vz)
* similar simplification as in updatePosition applies

Reply:
Actually, as you are using initialization by file, this should not be done in C++.
It's not a clean style to parse some of the IC from file, but then also do some further init in C++ on top.
You should very easily be able to correctly initialize the velocities in the external program that generated the file,
then you don't have to mix two different styles of initialization.